Source/santad/SNTNotificationQueue.m

```diff
@@ -79,10 +83,17 @@ - (void)flushQueue {
   @synchronized(self.pendingNotifications) {
     NSMutableArray *postedNotifications = [NSMutableArray array];
     for (NSDictionary *d in self.pendingNotifications) {
+      void (^reply)(BOOL authenticated) = d[@"reply"];
+      if (reply == nil) {
+        // The reply block sent to the GUI cannot be nil.
+        reply = ^(BOOL _) {
```

Comment:
We should maybe revert to the old style and expect a block be passed in (then also add code to check for nil replies and bail). We previously needed to ensure we always copied blocks to ensure stack-based blocks got moved to the heap. We would presumably hit the same issue here.

Reply:
I've added a copy of the reply block. I like the idea of SNTNotificationQueue being able to tell whether there is a reply block even if we're not currently using it.

Reply:
This does do two copies now for blocks passed in, but not really a huge deal.